File: examples/ospf-kotlin-example/src/main/fuookami/ospf/kotlin/example/framework_demo/demo2/domain/aircraft/model/HatchDoor.kt
package fuookami.ospf.kotlin.example.framework_demo.demo2.domain.aircraft.model

import fuookami.ospf.kotlin.utils.math.*
import fuookami.ospf.kotlin.utils.physics.quantity.*

enum class DoorUbietyType {
    Front,
    AdjacentFront,
    Beside,
    Opposite,
    AdjacentBehind,
    Behind;
}

data class DoorUbiety(
    val type: DoorUbietyType,
    val sameSide: Boolean,
    val position: Position,
    val door: HatchDoor
)

data class HatchDoor(
    val location: DeckLocation,
    val besideBulk: Boolean,
    val noseDoor: Boolean,
    val lateralArm: Quantity<Flt64>,
    val frontArm: Quantity<Flt64>,
    val backArm: Quantity<Flt64>
)

File: ospf-rust-example/src/framework/demo2/domain/aircraft/model/hatch_door.rs
use super::deck::DeckLocation;
use ospf_rust_quantities::quantity::Quantity;
use ospf_rust_quantities::unit::Unit;

/// 舱门 / Hatch door (对齐 Kotlin HatchDoor)
#[derive(Debug, Clone)]
pub struct HatchDoor {
    pub name: String,
    pub location: DeckLocation,
    pub beside_bulk: bool,
    pub nose_door: bool,
    pub lateral_arm: Quantity<f64, Unit>,
    pub front_arm: Quantity<f64, Unit>,
    pub back_arm: Quantity<f64, Unit>,
}
